toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -493,6 +494,15 @@ func PopulateInstallRoot(installChroot *safechroot.Chroot, packagesToInstall []s
 		return
 	}
 
+	if len(config.Users) > 0 || len(config.Groups) > 0 {
```

Comment:
Maybe something along these lines? Or failing that ensure that `shadow-utils` is in the install list in code.

```suggestion
	// imageconfigvalidator should have ensured that we intend to install shadow-utils, so we ca go ahead and do that here.
	if len(config.Users) > 0 || len(config.Groups) > 0 {
```

Reply:
Or
```suggestion
	if len(config.Users) > 0 || len(config.Groups) > 0 {
		if ! sliceutils.ContainsValue(packagesToInstall, "shadow-utils") {
			err = fmt.Errorf("shadow-utils package is required when setting users or groups")
			return
		}
```

Reply:
Done.